File: dart-jaguar/lib/model/emoji_reaction_base_user.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';


part 'emoji_reaction_base_user.jser.dart';

class EmojiReactionBaseUser {
   /* ID of the user.  */
  @Alias('id', isNullable: false,  )
  final int id;
   /* Email of the user.  */
  @Alias('email', isNullable: false,  )
  final String email;
   /* Full name of the user.  */
  @Alias('full_name', isNullable: false,  )
  final String fullName;
   /* Whether the user is a mirror dummy.  */
  @Alias('is_mirror_dummy', isNullable: false,  )
  final bool isMirrorDummy;
  

  EmojiReactionBaseUser(
      

{
     this.id = null,  
     this.email = null,  
     this.fullName = null,  
     this.isMirrorDummy = null 
    
    }
  );

  @override
  String toString() {
    return 'EmojiReactionBaseUser[id=$id, email=$email, fullName=$fullName, isMirrorDummy=$isMirrorDummy, ]';
  }
}

@GenSerializer(nullableFields: true)
class EmojiReactionBaseUserSerializer extends Serializer<EmojiReactionBaseUser> with _$EmojiReactionBaseUserSerializer {

}

File: scalaz/src/main/scala/org/openapitools/client/api/EmojiReactionBaseUser.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import EmojiReactionBaseUser._

case class EmojiReactionBaseUser (
  /* ID of the user.  */
  id: Option[Integer],
/* Email of the user.  */
  email: Option[String],
/* Full name of the user.  */
  fullName: Option[String],
/* Whether the user is a mirror dummy.  */
  isMirrorDummy: Option[Boolean])

object EmojiReactionBaseUser {
  import DateTimeCodecs._

  implicit val EmojiReactionBaseUserCodecJson: CodecJson[EmojiReactionBaseUser] = CodecJson.derive[EmojiReactionBaseUser]
  implicit val EmojiReactionBaseUserDecoder: EntityDecoder[EmojiReactionBaseUser] = jsonOf[EmojiReactionBaseUser]
  implicit val EmojiReactionBaseUserEncoder: EntityEncoder[EmojiReactionBaseUser] = jsonEncoderOf[EmojiReactionBaseUser]
}
